storage/innobase/btr/btr0bulk.cc

```diff
@@ -62,6 +62,10 @@ PageBulk::init()
 		the allocation order, and we will always generate redo log
 		for page allocation, even when creating a new tablespace. */
 		mtr_start(&alloc_mtr);
+		if (dict_index_get_space(m_index)
```

Comment:
It is upstream bug http://bugs.mysql.com/bug.php?id=39833. Percona feature expanded fast index creation has resulted, as a side effect, in FIC enabled for temp tables at least in some cases, partially fixing that bug: https://bugs.launchpad.net/percona-server/+bug/999641. In lower versions it mostly works as-is, in 5.7 more redo log mode (and DD, IIRC) handling is required. Added a comment. Will not submit upstream, this is a partial patch here.